geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZInterStoreIntegrationTest.java

```diff
@@ -78,6 +78,32 @@ public void shouldError_givenWrongKeyType() {
         STRING_KEY, KEY1)).hasMessage("WRONGTYPE " + RedisConstants.ERROR_WRONG_TYPE);
   }
 
+  @Test
+  public void shouldError_givenTwoSortedSetKeysAndWrongKeyType() {
+    Map<String, Double> scores = buildMapOfMembersAndScores(1, 5);
+    Map<String, Double> nonIntersectionScores = buildMapOfMembersAndScores(6, 10);
+    jedis.zadd(KEY1, scores);
+    jedis.zadd(KEY2, nonIntersectionScores);
+
+    final String STRING_KEY = "{tag1}stringKey";
+    jedis.set(STRING_KEY, "value");
+
+    assertThatThrownBy(() -> jedis.zinterstore(NEW_SET, KEY1, KEY2,
+        STRING_KEY)).hasMessageContaining(RedisConstants.ERROR_WRONG_TYPE);
+  }
+
+  @Test
+  public void shouldError_givenWrongKeyTypeAndTwoSortedSetKeys() {
```

Comment:
This test name might be a little clearer as something like "shouldReturnWrongTypeError_givenNonSortedSetKeyAsFirstSourceKey". Also, this test case duplicates the test "shouldError_givenWrongKeyType" but is clearer than that one, so that one should probably be removed.